xbmc/addons/AddonCallbacks.cpp

```diff
@@ -33,26 +34,29 @@ namespace ADDON
 
 CAddonCallbacks::CAddonCallbacks(CAddon* addon)
 {
-  m_addon       = addon;
-  m_callbacks   = new AddonCB;
-  m_helperAddon = NULL;
-  m_helperADSP  = NULL;
-  m_helperGUI   = NULL;
-  m_helperPVR   = NULL;
-  m_helperCODEC = NULL;
-
-  m_callbacks->libBasePath           = strdup(CSpecialProtocol::TranslatePath("special://xbmcbin/addons").c_str());
-  m_callbacks->addonData             = this;
-  m_callbacks->AddOnLib_RegisterMe   = CAddonCallbacks::AddOnLib_RegisterMe;
-  m_callbacks->AddOnLib_UnRegisterMe = CAddonCallbacks::AddOnLib_UnRegisterMe;
-  m_callbacks->ADSPLib_RegisterMe    = CAddonCallbacks::ADSPLib_RegisterMe;
-  m_callbacks->ADSPLib_UnRegisterMe  = CAddonCallbacks::ADSPLib_UnRegisterMe;
-  m_callbacks->CODECLib_RegisterMe   = CAddonCallbacks::CODECLib_RegisterMe;
-  m_callbacks->CODECLib_UnRegisterMe = CAddonCallbacks::CODECLib_UnRegisterMe;
-  m_callbacks->GUILib_RegisterMe     = CAddonCallbacks::GUILib_RegisterMe;
-  m_callbacks->GUILib_UnRegisterMe   = CAddonCallbacks::GUILib_UnRegisterMe;
-  m_callbacks->PVRLib_RegisterMe     = CAddonCallbacks::PVRLib_RegisterMe;
-  m_callbacks->PVRLib_UnRegisterMe   = CAddonCallbacks::PVRLib_UnRegisterMe;
+  m_addon             = addon;
```

Comment:
or at least make it a separate commit. and of course omit it from the PR, because there's nothing sexier than a pr with 1740 lines added, 2 removed ;)